applications/ShapeOptimizationApplication/custom_utilities/optimization_utilities.h

```diff
@@ -182,7 +182,7 @@ class OptimizationUtilities
         double norm_2_dCds_i = 0.0;
         for (auto & node_i : mrDesignSurface.Nodes())
         {
-        	array_3d& dCds_i = node_i.FastGetSolutionStepValue(MAPPED_CONSTRAINT_SENSITIVITY);
+        	array_3d& dCds_i = node_i.FastGetSolutionStepValue(DC1DX_MAPPED);
```

Comment:
I think it would pay of to put the `penalized_projection` functions in a separate file.

Reply:
I agree. Perspectively all operations dedicated to a special algorithm should be separately implemented and assigned to the corresponding algorithm. Such a dependency should be also properly reflected by the naming and location of the classes and files. As this is a bit more effort, I would better see this in a separate PR. Nevertheless, for now, I split the c++ implementations related to the individual algorithms.